src/o365/spo/commands/site/site-rename.ts

```diff
@@ -0,0 +1,285 @@
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let spoAdminUrl: string = "";
+    let options = args.options;
+
+
+    this
+      .getSpoAdminUrl(cmd, this.debug)
+      .then((_spoAdminUrl: string): Promise<FormDigestInfo> => {
+        spoAdminUrl = _spoAdminUrl;
+
+        return this.getRequestDigest(spoAdminUrl);
+      })
+      .then((res: FormDigestInfo): Promise<string> => {
+        this.context = res;
+        if (this.verbose) {
+          cmd.log(`Scheduling rename job...`);
+        }
+
+        let optionsBitmask = 0;
+        if (options.suppressMarketplaceAppCheck) {
+          optionsBitmask = optionsBitmask | 8;
+        }
+        
+        if (options.suppressWorkflow2013Check) {
+          optionsBitmask = optionsBitmask | 16;
+        }
+
+        let requestOptions = {
+          "SourceSiteUrl":options.siteUrl,
+          "TargetSiteUrl":options.newSiteUrl,
+          "TargetSiteTitle":options.newSiteTitle||null,
+          "Option": optionsBitmask,
+          "Reserve":null,
+          "SkipGestures":null,
+          "OperationId":"00000000-0000-0000-0000-000000000000"
+        };
+
+        const postData: any = {
+          url: `${spoAdminUrl}/_api/SiteRenameJobs?api-version=1.4.7`,
+          headers: {
+            'X-RequestDigest': this.context.FormDigestValue,
+            'Content-Type': 'application/json'
+          },
+          json: true,
+          body: requestOptions
+        };
+        
+        return request.post(postData);
+      })
+      .then((res: any): Promise<void> => {
+          if (options.verbose) {
+            cmd.log(res);
+          }
+          
+          this.operationData = res;
+
+          if (this.operationData.JobState && this.operationData.JobState=="Error")
+          {
+            return Promise.reject(this.operationData.ErrorDescription);
+          }
+
+          const isComplete: boolean = (this.operationData.JobState && this.operationData.JobState==="Success");
+          if (!options.wait || isComplete) {
+            return Promise.resolve();
+          }
+          
+          
+
+          return new Promise<void>((resolve: () => void, reject: (error: any) => void): void => {
+            
+              this.waitForRenameCompletion(
+                this,
+                true,
+                spoAdminUrl,
+                options.siteUrl, 
+                resolve, 
+                reject, 
+                cmd, 
+                0
+              );
+  
+          })
+          
+      }).then((): void => {
+        if (args.options.output === 'json') {
+          this.operationData.JobState = "Success";
+          cmd.log(this.operationData);
+        }
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb()
+      }, (err: any): void => this.handleRejectedPromise(err, cmd, cb));
```

Comment:
Because we specified that the requests returns JSON data `json: true`, we should use `handleRejectedODataJsonPromise` to handle a faulty promise correctly